hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmDBAccessIdInfo.java

```diff
@@ -42,16 +47,23 @@
    * Only effective if isAdmin is true.
    */
   private final boolean isDelegatedAdmin;
+  /**
+   * Role names of the user (that this access ID is assigned to) in this tenant.
+   * e.g. OzoneConsts.TENANT_ROLE_USER, OzoneConsts.TENANT_ROLE_ADMIN,
+   *      or other custom role names.
+   */
+  private final Set<String> roleIds;
```

Comment:
I use "Id" suffix here just to be consistent with others:

```
tenantId = tenant name
policyId = policy name
roleId   = role name
```

Yeah it is a bit misleading as none of those "Id"s means "number".. 

Maybe change all of them (back) to a "Name" suffix?

Reply:
For PolicyID, we do use the number, but stored as String: 

https://github.com/apache/ozone/blob/8ecff6fd1a5842e8c81c125fd1d5aa729b7c7a47/hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/multitenant/RangerAccessPolicy.java#L72

If we change it to `long` we should change the type in `OmDBTenantState` as well.

Reply:
Seems odd that we use name for role and ID for policy. I think we should standardize and use either IDs for role and policy, or names for role and policy. When a policy is created, roles can be created in the same operation, however you will only get the policy ID back. This means we need to track roles by name, so I think we should track policies by name as well. I think this will make error handling, retries, and tests for existence easier as well.

I am currently updating the `MultiTenantAccessController` to operate on names instead of IDs.